docs/config.md

```diff
@@ -122,18 +124,304 @@ model_names: <comma-delimited-string|list>
+```yaml
+perf_latency:
+    max: 100
 ```
 
-All the flags supported in the CLI, are supported in the configuration file
-too.
+The config above will filter out the results that their `perf_latency` is larger than
+100 milliseconds.
+
+3. `gpu_used_memory`: Using this key you can specify the maximum GPU memory used by the model.
+The unit for this field is megabytes.
+```yaml
+gpu_used_memory:
+    max: 200
+```
+
+The config above will filter out the results that their `gpu_used_memory` is larger than
+200 MBs.
+
+You can also use these keys together:
+```yaml
+gpu_used_memory:
+    max: 200
+perf_latency:
+    max: 100
+```
+
+This will filter out the results that their `gpu_used_memory` is larger than 200 MBs and
+their latency is larger than 100 milliseconds.
+
+The values described above can be specified both globally and on a per model basis.
+The global example looks like below:
+
+```yaml
+model_repository: /path/to/model-repository
+model_names:
+  - model_1
+  - model_2
+
+constraints:
+    gpu_used_memory:
+        max: 200
+```
+
+In the global mode, the constraint specified will be enforced on every model.
+If you have different constraints for each model, you can use the per model version:
+
+```yaml
+model_repository: /path/to/model-repository
+model_names:
+  model_1:
+    constraints:
+        gpu_used_memory:
+            max: 200
+  model_2:
+    constraints:
+        latency:
+            max: 50
+```
+
+## `<objective>`
+
+Objectives allow you to specify the sorting criteria for the final results.
+The fields below are supported under this object type:
+
+1. `perf_throughput`
+2. `perf_latency`
+3. `gpu_used_memory`
+4. `gpu_free_memory`
+5. `gpu_utilization`
+6. `cpu_used_ram`
+7. `cpu_available_ram`
+
+An example `objectives` that will sort the results by throughput looks like below:
+
+```yaml
+objectives:
+- perf_throughput
+```
+
+If you want to sort the results by latency, `objectives` should look like:
+
+```yaml
+objectives:
+- perf_latency
+```
+### Weighted Objectives
+
+In addition to the mode discussed above, you provide multiple values in
+the objectives key. For example:
+
+```yaml
+objectives:
+- perf_latency
+- perf_throughput
+```
+
+The above config will multiply obtained throughput and latency for each
+measurement by 0.5 (i.e. 0.5 * latency + 0.5 * throughput) and use this as
+the "score" to sort the measurements according to. This mode can be useful
+if you have a more advanced criteria upon which you want to sort the results.
+
+An extension of the above `objectives` is explicitly specifying the weights.
+For example:
+```yaml
+objectives:
+    perf_latency: 10
+    perf_throughput: 15
+```
+
+The score for each measurement will be a weighted average using the weights specified here.
+
+Similar to `<constraint>`, `<objective>` can be specified both globally and on a
+per model basis.
+
+## `<parameter>`
+
+This type can contain `concurrency` or `batch_sizes` as the possible keys. You
```

Comment:
"A user can specify a range of test configurations that Model Analyzer will profile over. The possible configuration parameters are `concurrency` and `batch_size`. One or more parameters are specified per model only. Parameters cannot be globally specified"

Reply:
Could also start a table here. Tables are useful and make it easy to document new configurations